src/index.ts

```diff
@@ -148,6 +152,14 @@ export default (extendedPinoOptions?: ExtendedPinoOptions): PinoLambdaLogger =>
       logger.level = configuredLevel;
     }
 
+    // handle custom request level mixins
+    if (pinoOptions.requestMixin) {
+      const result = pinoOptions.requestMixin(event, context);
+      for (const key in result) {
+        ctx[key] = result[key] as string;
+      }
```

Comment:
How does this look as a safer alternative to the `string` cast?

```suggestion
      for (const [key, value] of Object.entries(result)) {
        if (value) {
          // Set the value on the context if the result value is defined
          ctx[key] = value;
        } else if (ctx[key]) {
          // Unset the existing context value if the result value is undefined
          delete ctx[key]
        }
      }
```

Reply:
Oh that's a good point–maybe a comment above the original with that context would be enough? Especially `When the JSON serializer runs, but default it omits undefined properties.`